pkg/operator/operator.go

```diff
@@ -381,20 +378,20 @@ func (o *Operator) handleEvent(obj interface{}) {
 	o.enqueue(cmoConfigMap)
 }
 
-func (o *Operator) worker() {
-	for o.processNextWorkItem() {
+func (o *Operator) worker(ctx context.Context) {
+	for o.processNextWorkItem(ctx) {
```

Comment:
@simonpasquier I am guessing that if the context gets cancelled, the next work item be processed. Could you please confirm my hunch?

Reply:
good catch! yes I agree that we should verify whether the context has been cancelled before processing the next item.

Reply:
at the same time, `Run()` currently doesn't wait for any of the goroutines to stop before exiting so it would need to be addressed not only for `worker()` IMHO. I'm ok with having a follow-up PR to "fix" this.